app/Kconfig

```diff
@@ -248,6 +248,39 @@ config ZMK_RGB_UNDERGLOW_ON_START
 #ZMK_RGB_UNDERGLOW
 endif
 
+menuconfig ZMK_BACKLIGHT
+	bool "LED backlight"
+	select PWM
+	select LED
+	select LED_PWM
+
+if ZMK_BACKLIGHT
+
+config ZMK_BACKLIGHT_LEVELS
+	int "Number of brightness levels"
+	range 2 100
+	default 6
+
+config ZMK_BACKLIGHT_LEVEL_START
+	int "Default brightness level"
+	range 1 ZMK_BACKLIGHT_LEVELS
```

Comment:
Adding `range 0 (ZMK_BACKLIGHT_LEVELS-1)` to the Kconfig file generates an error, so the best compromise I have found is to use values in range [1, ZMK_BACKLIGHT_LEVELS] in the config file and values in range [0, ZMK_BACKLIGHT_LEVELS-1] in the `.c` file. Do you know any workaround?

Reply:
The problem with that is you are initializing the brightness to `CONFIG_ZMK_BACKLIGHT_LEVEL_START - 1`, which will be 0 if you set an initial brightness of 1. Two solutions I can think of:

1. Set the range to be 1-100, and initialize the brightness to `MIN(CONFIG_ZMK_BACKLIGHT_LEVEL_START, CONFIG_ZMK_BACKLIGHT_LEVELS - 1)` to silently ignore values that are too large and clamp them.
2. Set the range to be 1-100, and initialize the brightness to `CONFIG_ZMK_BACKLIGHT_LEVEL_START`. Add a `BUILD_ASSERT(CONFIG_ZMK_BACKLIGHT_LEVEL_START < CONFIG_ZMK_BACKLIGHT_LEVELS, "CONFIG_ZMK_BACKLIGHT_LEVEL_START is too large")`  to cause a build error if the value is too large.
3. Keep the range as it is, but then add something like
  ```c
  #define MIN_BRIGHTNESS 1
  ```
  and then treat `state.brightness` as an offset above the minimum brightness instead of an absolute brightness.

I think options 1 and 2 would make the code simpler, because then you don't have to remember that 0 actually means 1.

Reply:
I have replaced `range 1 ZMK_BACKLIGHT_LEVELS` with `range 2 ZMK_BACKLIGHT_LEVELS` in the Kconfig file, this should fix that.